app/src/app/[lng]/[inventory]/data/[step]/ActivityDataTab.tsx

```diff
@@ -49,9 +50,9 @@ const activityDataUnits: Record<string, string[]> = {
 ];*/
 
 export function determineEmissionsFactorType(factor: EmissionsFactorData) {
-  let sourceName = factor.dataSources
-    ? factor.dataSources[0].datasetName || "Unknown data source"
-    : "Unknown data source";
+  const sourceName =
+    getTranslationFromDict(factor.dataSources?.[0]?.datasetName || {}) ||
+    "Unknown data source";
```

Comment:
Use TypeScript's `??` default operator if possible as there are less weird type coercions with it